src/core/gltf2importer/gltf2parser.cpp

```diff
@@ -476,6 +506,55 @@ Qt3DCore::QEntity *GLTF2Parser::parse(const QByteArray &jsonData, const QString
     return gltfSceneEntity;
 }
 
+Qt3DCore::QEntity *GLTF2Parser::parse(QIODevice *stream, const QString &basePath, const QString &filename)
+{
+    QByteArray jsonData;
+    while (!stream->atEnd()) {
+        struct ChunkHeader {
+            quint32 chunkLength;
+            quint32 chunkType;
+        } chunkHeader;
+        if (stream->read(reinterpret_cast<char *>(&chunkHeader), sizeof(chunkHeader)) != sizeof(chunkHeader)) {
+            qCWarning(kuesa()) << "Malformed chunck in glb file";
+            return nullptr;
+        }
+
+        auto data = stream->read(chunkHeader.chunkLength);
+        if (data.size() != chunkHeader.chunkLength) {
+            qCWarning(kuesa()) << "Malformed chunck in glb file";
+            return nullptr;
+        }
+
+        if (chunkHeader.chunkType == 0x4E4F534A) {
```

Comment:
move the hardcoded value to a const variable like we do for JSON keys?

Reply:
done